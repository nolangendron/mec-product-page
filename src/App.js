import React, { useState } from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import { LandingPage } from './LandingPage/LandingPage'
import { ProductPage } from './ProductPage/ProductPage'
import { makeApiCall } from '../src/utils/apiCall'

function App() {
  const [searchQuery, setSearchQuery] = useState('')
  const [currentSearch, setCurrentSearch] = useState('')
  const [fetchSuccess, setFetchSuccess] = useState(false)
  const [products, setProducts] = useState([])
  const [currentPage, setCurrentPage] = useState(1)
  const [productsPerPage] = useState(12)

  const indexOfLastProduct = currentPage * productsPerPage
  const indexOfFirstProduct = indexOfLastProduct - productsPerPage
  const currentProducts = products.slice(
    indexOfFirstProduct,
    indexOfLastProduct
  )

  const toggleFetchSuccess = () => {
    setFetchSuccess(false)
  }
  const handleQueryChange = (e) => {
    const value = e.target.value
    setSearchQuery(value)
  }

  const handleSearch = (e) => {
    e.preventDefault()
    setCurrentSearch(searchQuery)
    if (searchQuery.length === 0) {
      return
    } else {
      makeApiCall(searchQuery).then((results) => {
        const data = results
        const { products } = data
        setProducts(products)
        setCurrentPage(1)
      })
      setFetchSuccess(true)
    }
  }

  const paginate = (pageNumber) => setCurrentPage(pageNumber)
  const paginateArrow = (arrow) => {
    arrow === 'forward'
      ? setCurrentPage((prevState) => prevState + 1)
      : setCurrentPage((prevState) => prevState - 1)
  }

  return (
    <Router>
      <Switch>
        <Route path={`/products`}>
          <ProductPage
            searchQuery={searchQuery}
            currentSearch={currentSearch}
            handleQueryChange={handleQueryChange}
            handleSearch={handleSearch}
            products={products}
            currentProducts={currentProducts}
            paginate={paginate}
            paginateArrow={paginateArrow}
            currentPage={currentPage}
            productsPerPage={productsPerPage}
            toggleFetchSuccess={toggleFetchSuccess}
          />
        </Route>
        <Route path="/" exact>
          <LandingPage
            currentSearch={currentSearch}
            searchQuery={searchQuery}
            handleQueryChange={handleQueryChange}
            handleSearch={handleSearch}
            fetchSuccess={fetchSuccess}
          />
        </Route>
        <Route render={() => <h2>Not Found</h2>} />
      </Switch>
    </Router>
  )
}

export default App
