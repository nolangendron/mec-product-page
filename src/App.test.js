import React from 'react'
import { render } from '@testing-library/react'
import { shallow } from 'enzyme'
import App from './App'

it('should take a snapshot of App', () => {
  const { asFragment } = render(<App />)
  expect(asFragment(<App />)).toMatchSnapshot()
})

it('renders without crashing', () => {
  shallow(<App />)
})
