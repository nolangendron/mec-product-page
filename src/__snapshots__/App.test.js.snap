// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of App 1`] = `
<DocumentFragment>
  <div
    class="css-0"
  >
    <form>
      <input
        placeholder="Search for product"
        value=""
      />
    </form>
  </div>
  <section />
</DocumentFragment>
`;
